### DISCUSSION TITLE
Grid option in the editor.

### DISCUSSION THREAD

**Discussion by jaspreet-singh-1999** [GrapesJS/grapesjs#5569] [Q&A]
hi,
I want grid in the editor and also want to add its trait how can i do ??

**Comment by mosh-tudor** [comment]
@jaspreet-singh-1999 

As for traits, you can do everything using the default functionality.

I managed to achieve this result:

<img width="301" alt="Screenshot 2023-12-21 at 12 30 11" src="https://github.com/GrapesJS/grapesjs/assets/668451/5d458aa5-3e68-4e6b-b532-00d4b0abff1a">

<img width="302" alt="Screenshot 2023-12-21 at 12 33 06" src="https://github.com/GrapesJS/grapesjs/assets/668451/6c92f4ec-5b11-4093-8d67-2b1bcc06d3f7">

---

I think the key to success here is the "requires" and "requiresParent" properties, which can be used to hide unnecessary fields., 

```js
editor.StyleManager.addSector('layout', {
    name: 'Layout',
    open: true,
    properties: [
      {
        label: 'Display',
        property: 'display',
        default: 'block',
        type: 'radio',
        list: [
          { value: 'block' },
          { value: 'flex' },
          { value: 'grid' },
          { value: 'inline-block' },
          { value: 'inline' },
          { value: 'none' },
        ],
      },
      {
        label: 'Direction',
        property: 'flex-direction',
        requires: { display: ['flex', 'inline-flex'] }, // <--------
        type: 'select',
        default: 'row',
        full: true,
        list: [...],
      },
      {
        label: 'Align',
        property: 'align-items',
        requires: {
          display: ['flex', 'inline-flex'], 'flex-direction': ['', 'row', 'row-reverse'], // <--------
        },
        default: 'stretch',
        type: 'radio',
        list: [...],
      },
  ...
...
```

**Comment by jaspreet-singh-1999** [comment]
Not show properly,
 
![Screenshot from 2023-12-21 18-13-34](https://github.com/GrapesJS/grapesjs/assets/120703382/eacabbe2-673b-4655-bdd9-13bbf87c7af7)

**Reply by mosh-tudor** [reply]
It's better to start by reading the documentation, it will save you a lot of time:

https://grapesjs.com/docs/getting-started.html

`+` To get help you need to post the code on JSFiddle. Via screenshot, I don't think anyone will find what you did wrong ✌️.

**Reply by artf** [reply]
well said @mosh-tudor 🥇
